### PR TITLE
test: generate e-waybill for Delivery Note and Purchase Invoice

### PR DESCRIPTION
**Pull request by DaizyModi**
Covered cases for:
1. Delivery Note
    a. DN with Same GSTIN
    b. DN with different GSTIN
    c. Return DN with same GSTIN
    d. Return DN with different GSTIN
2. Purchase Invoice
    a. Validate Bill No for Registered Supplier
    b. e-waybill for Purchase Invoice
    c. e-waybill for Purchase Return

### CODE REVIEW

File: india_compliance/gst_india/data/test_e_waybill.json [hunk 1]
```diff
@@ -352,5 +352,514 @@
...
+            "TotNonAdvolVal": 0,
+            "actFromStateCode": 24,
+            "actToStateCode": 24,
+            "cessValue": 0,
+            "cgstValue": 0,
+            "docDate": "08/08/2023",
+            "docNo": "test_invoice_no",
+            "docType": "INV",
+            "fromAddr1": "Test Address - 10",
+            "fromGstin": "05AAACG2140A1ZL",
```
Comment (vorasmit) [inline india_compliance/gst_india/data/test_e_waybill.json]: gstin for unregistered?

File: india_compliance/gst_india/data/test_e_waybill.json [hunk 2]
```diff
@@ -352,5 +352,514 @@
...
+        },
+        "request_data": {
+            "OthValue": 0.0,
+            "TotNonAdvolVal": 0,
+            "actFromStateCode": 24,
+            "actToStateCode": 24,
+            "cessValue": 0,
+            "cgstValue": 0,
+            "docDate": "08/08/2023",
+            "docNo": "test_invoice_no",
```
Comment (vorasmit) [inline india_compliance/gst_india/data/test_e_waybill.json:667]: send bill_no in kwargs and validate it here.

File: india_compliance/gst_india/data/test_e_waybill.json [hunk 3]
```diff
@@ -352,5 +352,514 @@
...
+            },
+            "success": true
+        }
+    },
+    "pi_data_for_registered_supplier": {
+        "kwargs": {
+            "supplier": "_Test Registered Supplier",
+            "qty": 1000,
+            "distance": 10,
+            "mode_of_transport": "Road",
```
Comment (vorasmit) [inline india_compliance/gst_india/data/test_e_waybill.json:727]: add bill_no and assert its data.

File: india_compliance/gst_india/data/test_e_waybill.json (hunk 2 above)
Comment (DaizyModi) [reply india_compliance/gst_india/data/test_e_waybill.json:667]: In test case (sandbox mode), not needed

**Pull request merged**